### PR TITLE
Make ConfirmationModal more general

### PR DESCRIPTION
**Pull request by sunny-viktoryia**
prepare the existing ConfirmationModal for the NordBeavers team

### CODE REVIEW

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 1]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
...
+      onHide={hide}
+      centered={centered}
+    >
       <Modal.Body>
-        <div className="confirmation-modal">
-          {hasHeader && <h2 className="confirm-header">{header}</h2>}
+        <div>
+          {hasHeader && <h3 className="confirm-header">{header}</h3>}
           <div className="confirm-message">{message}</div>
           <div className="confirmation-buttons">
```
Comment (goran-peoski-work) [inline src/components/atoms/ConfirmationModal/ConfirmationModal.tsx:62]:
```suggestion
          <div className="ConfirmationModal__buttons confirmation-buttons">
```

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 2]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
...
+    <Modal
+      className="confirmation-modal"
+      show={isShown}
+      onHide={hide}
+      centered={centered}
+    >
       <Modal.Body>
-        <div className="confirmation-modal">
-          {hasHeader && <h2 className="confirm-header">{header}</h2>}
+        <div>
```
Comment (goran-peoski-work) [inline src/components/atoms/ConfirmationModal/ConfirmationModal.tsx:59]:
```suggestion
        <div className="ConfirmationModal__content">
```

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 3]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
...
+          {hasHeader && <h3 className="confirm-header">{header}</h3>}
           <div className="confirm-message">{message}</div>
```
Comment (goran-peoski-work) [inline src/components/atoms/ConfirmationModal/ConfirmationModal.tsx:61]:
```suggestion
          <div className="ConfirmationModal__message confirm-message">{message}</div>
```

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 4]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
...
+      className="confirmation-modal"
+      show={isShown}
+      onHide={hide}
+      centered={centered}
+    >
       <Modal.Body>
-        <div className="confirmation-modal">
-          {hasHeader && <h2 className="confirm-header">{header}</h2>}
+        <div>
+          {hasHeader && <h3 className="confirm-header">{header}</h3>}
```
Comment (goran-peoski-work) [inline src/components/atoms/ConfirmationModal/ConfirmationModal.tsx:60]: Should probably be part of `Modal.Title` without the `h3` that most likely breaks the semantics of h* elements just for the sake of visual appearance

```suggestion
          {hasHeader && <h3 className="ConfirmationModal__header confirm-header">{header}</h3>}
```

File: src/components/atoms/ConfirmationModal/ConfirmationModal.scss [hunk 5]
```diff
@@ -1,24 +1,14 @@
 @import "scss/constants.scss";
 
 .confirmation-modal {
+  .modal-content {
+    background: $grey--modal-background;
```
Comment (goran-peoski-work) [inline src/components/atoms/ConfirmationModal/ConfirmationModal.scss:5]: Suggested name in `scss/constants.scss`:


```suggestion
    background: $modal-background--confirm;
```

**Review by goran-peoski-work — COMMENTED** [review]
Some comments on class names and a color constant

File: src/scss/constants.scss [hunk 6]
```diff
@@ -33,6 +33,7 @@ $light-grey: #292929;
 $light-grey--lightest: #acadb0;
 $concrete: #a3a6ab;
 $concrete--dark: #9f9fa0;
+$grey--modal-background: #232225;
```
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/scss/constants.scss]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 7]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
...
-              Yes
-            </Button>
+            <ButtonNG className="cancel-button" onClick={cancel}>
+              {no}
+            </ButtonNG>
+            <ButtonNG
+              className="confirm-button"
+              variant="primary"
+              onClick={confirm}
+            >
```
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx]: removed the classes as we don't have css for them
fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx [hunk 8]
```diff
@@ -41,18 +49,27 @@ export const ConfirmationModal: FC<ConfirmationModalProps> = ({
   const isShown = show !== undefined ? show : isVisible;
 
   return (
-    <Modal show={isShown} onHide={hide}>
+    <Modal
+      className="confirmation-modal"
```
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx (hunk 1 above)
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx (hunk 2 above)
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx:59]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx (hunk 3 above)
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.tsx (hunk 4 above)
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.tsx]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/ConfirmationModal/ConfirmationModal.scss (hunk 5 above)
Comment (sunny-viktoryia) [reply src/components/atoms/ConfirmationModal/ConfirmationModal.scss]: fixed here: https://github.com/sparkletown/sparkle/pull/2131/commits/9b9f6c1a6dd9bc6c13b69121718d28b857ec4bf4

File: src/components/atoms/AccessDeniedModal/AccessDeniedModal.tsx [hunk 9]
```diff
@@ -1,50 +0,0 @@
-import React, { useCallback, useState } from "react";
```
Comment (sunny-viktoryia) [inline src/components/atoms/AccessDeniedModal/AccessDeniedModal.tsx:1]: removed the component `AccessDeniedModal` as it's not used anywhere.
We should better use `ConfirmationModal`

**Review by goran-peoski-work — APPROVED** [review]
LGTM

**Pull request merged**